{"uid":"ee905e68d39f99c0","name":"test_database_relation","fullName":"tests.integration.test_charm#test_database_relation","historyId":"a46dd5fec5170ae1b5112e2bb246aae9","time":{"start":1726798988938,"stop":1726799891358,"duration":902420},"description":"Test the database relation.","descriptionHtml":"<p>Test the database relation.</p>\n","status":"broken","statusMessage":"asyncio.exceptions.TimeoutError","statusTrace":"Traceback (most recent call last):\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/juju/utils.py\", line 130, in _block\n    await jasyncio.sleep(wait_period)\n  File \"/usr/lib/python3.10/asyncio/tasks.py\", line 605, in sleep\n    return await future\nasyncio.exceptions.CancelledError\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/lib/python3.10/asyncio/tasks.py\", line 456, in wait_for\n    return fut.result()\nasyncio.exceptions.CancelledError\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 341, in from_call\n    result: Optional[TResult] = func()\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 262, in <lambda>\n    lambda: ihook(item=item, **kwds), when=when, reraise=reraise\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 493, in __call__\n    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 115, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 152, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 114, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 177, in pytest_runtest_call\n    raise e\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 169, in pytest_runtest_call\n    item.runtest()\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 1788, in runtest\n    self.ihook.pytest_pyfunc_call(pyfuncitem=self)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 493, in __call__\n    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 115, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 152, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 114, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 194, in pytest_pyfunc_call\n    result = testfunction(**testargs)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 532, in inner\n    _loop.run_until_complete(task)\n  File \"/usr/lib/python3.10/asyncio/base_events.py\", line 649, in run_until_complete\n    return future.result()\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/tests/integration/test_charm.py\", line 80, in test_database_relation\n    await ops_test.model.block_until(\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/juju/model.py\", line 1040, in block_until\n    await utils.block_until(done,\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/juju/utils.py\", line 131, in block_until\n    await jasyncio.shield(jasyncio.wait_for(_block(), timeout))\n  File \"/usr/lib/python3.10/asyncio/tasks.py\", line 458, in wait_for\n    raise exceptions.TimeoutError() from exc\nasyncio.exceptions.TimeoutError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":1,"retriesStatusChange":true,"beforeStages":[{"name":"event_loop","time":{"start":1726798988672,"stop":1726798988672,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"tmp_path_factory","time":{"start":1726798988672,"stop":1726798988672,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"ops_test","time":{"start":1726798988673,"stop":1726798988938,"duration":265},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"abort_on_fail","time":{"start":1726798988938,"stop":1726798988938,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"Test the database relation.","status":"broken","statusMessage":"asyncio.exceptions.TimeoutError","statusTrace":"Traceback (most recent call last):\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/juju/utils.py\", line 130, in _block\n    await jasyncio.sleep(wait_period)\n  File \"/usr/lib/python3.10/asyncio/tasks.py\", line 605, in sleep\n    return await future\nasyncio.exceptions.CancelledError\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/lib/python3.10/asyncio/tasks.py\", line 456, in wait_for\n    return fut.result()\nasyncio.exceptions.CancelledError\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 341, in from_call\n    result: Optional[TResult] = func()\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 262, in <lambda>\n    lambda: ihook(item=item, **kwds), when=when, reraise=reraise\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 493, in __call__\n    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 115, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 152, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 114, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 177, in pytest_runtest_call\n    raise e\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 169, in pytest_runtest_call\n    item.runtest()\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 1788, in runtest\n    self.ihook.pytest_pyfunc_call(pyfuncitem=self)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 493, in __call__\n    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 115, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 152, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 114, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 194, in pytest_pyfunc_call\n    result = testfunction(**testargs)\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 532, in inner\n    _loop.run_until_complete(task)\n  File \"/usr/lib/python3.10/asyncio/base_events.py\", line 649, in run_until_complete\n    return future.result()\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/tests/integration/test_charm.py\", line 80, in test_database_relation\n    await ops_test.model.block_until(\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/juju/model.py\", line 1040, in block_until\n    await utils.block_until(done,\n  File \"/home/runner/work/mysql-router-k8s-operator/mysql-router-k8s-operator/.tox/integration/lib/python3.10/site-packages/juju/utils.py\", line 131, in block_until\n    await jasyncio.shield(jasyncio.wait_for(_block(), timeout))\n  File \"/usr/lib/python3.10/asyncio/tasks.py\", line 458, in wait_for\n    raise exceptions.TimeoutError() from exc\nasyncio.exceptions.TimeoutError","steps":[],"attachments":[{"uid":"d7c4ca68f79ce75","name":"log","source":"d7c4ca68f79ce75.txt","type":"text/plain","size":3716}],"parameters":[],"stepsCount":0,"shouldDisplayMessage":true,"attachmentsCount":1,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"abort_on_fail::0","time":{"start":1726799891364,"stop":1726799891364,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"ops_test::finalizer","time":{"start":1726799891365,"stop":1726799899163,"duration":7798},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::2","time":{"start":1726799899164,"stop":1726799899164,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_close_event_loop","time":{"start":1726799899164,"stop":1726799899164,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_provide_clean_event_loop","time":{"start":1726799899164,"stop":1726799899164,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"tag","value":"abort_on_fail"},{"name":"tag","value":"group(1)"},{"name":"tag","value":"asyncio"},{"name":"parentSuite","value":"tests.integration"},{"name":"suite","value":"test_charm"},{"name":"host","value":"fv-az842-181"},{"name":"thread","value":"11907-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.integration.test_charm"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[{"uid":"975d6a246ee1b50e","status":"unknown","time":{}}],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":0,"broken":50,"skipped":0,"passed":60,"unknown":3,"total":113},"items":[{"uid":"7045e1ce61696754","reportUrl":"../269//#testresult/7045e1ce61696754","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1726712614384,"stop":1726713517574,"duration":903190}},{"uid":"a76ed1ad1f7b97ef","reportUrl":"../267//#testresult/a76ed1ad1f7b97ef","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1726626111103,"stop":1726627013444,"duration":902341}},{"uid":"803971efa5a1ceb9","reportUrl":"../266//#testresult/803971efa5a1ceb9","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1726538376150,"stop":1726539278215,"duration":902065}},{"uid":"48758bd67ec8bbd1","reportUrl":"../264//#testresult/48758bd67ec8bbd1","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1726453813740,"stop":1726454716155,"duration":902415}},{"uid":"25c5111b9655bd97","reportUrl":"../263//#testresult/25c5111b9655bd97","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1726367548469,"stop":1726368450941,"duration":902472}},{"uid":"c237ef8e5a8ce4b5","reportUrl":"../262//#testresult/c237ef8e5a8ce4b5","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1726280413191,"stop":1726281315722,"duration":902531}},{"uid":"8e285d37f181b8b8","reportUrl":"../260//#testresult/8e285d37f181b8b8","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1726194229186,"stop":1726195132281,"duration":903095}},{"uid":"8f34df5729e787c7","reportUrl":"../259//#testresult/8f34df5729e787c7","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1726107773819,"stop":1726108676136,"duration":902317}},{"uid":"b808c1adbfc8f1b4","reportUrl":"../256//#testresult/b808c1adbfc8f1b4","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1726021336887,"stop":1726022239787,"duration":902900}},{"uid":"99022fbb50ae3adb","reportUrl":"../253//#testresult/99022fbb50ae3adb","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1725935004982,"stop":1725935908081,"duration":903099}},{"uid":"e7ea21118a941969","reportUrl":"../251//#testresult/e7ea21118a941969","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1725848695801,"stop":1725849598285,"duration":902484}},{"uid":"da3bef71e7a146ee","reportUrl":"../250//#testresult/da3bef71e7a146ee","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1725762429350,"stop":1725763331705,"duration":902355}},{"uid":"dbb4964f6b320fd6","reportUrl":"../249//#testresult/dbb4964f6b320fd6","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1725675595995,"stop":1725676498281,"duration":902286}},{"uid":"a5b9d5135bf6b4f0","reportUrl":"../247//#testresult/a5b9d5135bf6b4f0","status":"unknown","time":{}},{"uid":"6f978be00de9417e","reportUrl":"../245//#testresult/6f978be00de9417e","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1725502941099,"stop":1725503843620,"duration":902521}},{"uid":"f3302b519c9c9fce","reportUrl":"../244//#testresult/f3302b519c9c9fce","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1725416472514,"stop":1725417375097,"duration":902583}},{"uid":"3cb0f19d7ee672ab","reportUrl":"../243//#testresult/3cb0f19d7ee672ab","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1725329996392,"stop":1725330898461,"duration":902069}},{"uid":"cf75650e40e1bfc5","reportUrl":"../240//#testresult/cf75650e40e1bfc5","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1725243756420,"stop":1725244658368,"duration":901948}},{"uid":"9c6681450777bd25","reportUrl":"../239//#testresult/9c6681450777bd25","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1725158015009,"stop":1725158917130,"duration":902121}},{"uid":"34c1a98c35755a49","reportUrl":"../238//#testresult/34c1a98c35755a49","status":"broken","statusDetails":"asyncio.exceptions.TimeoutError","time":{"start":1725070881029,"stop":1725071783414,"duration":902385}}]},"tags":["asyncio","abort_on_fail","group(1)"]},"source":"ee905e68d39f99c0.json","parameterValues":[]}